aws-predictions/src/main/java/com/amplifyframework/predictions/aws/http/LivenessWebSocket.kt

```diff
@@ -98,6 +102,15 @@ internal class LivenessWebSocket(
     internal var clientStoppedSession = false
     val json = Json { ignoreUnknownKeys = true }
 
+    // Sending events to the websocket requires processing synchronously because we rely on proper ordered
```

Comment:
This pattern is used in `AWSCognitoAuthPlugin` as well. Can we replace both of these usages with a utility class that encapsulates some of the details? It'll make it easier to adjust this logic if needed.

```kotlin
class CoroutineQueue(
    consumingDispatcher: CoroutineDispatcher = Dispatchers.Default
) {
    private val sendEventScope = CoroutineScope(SupervisorJob() + consumingDispatcher)
    private val sendEventQueueChannel = Channel<Job>(capacity = Channel.UNLIMITED).apply {
        sendEventScope.launch {
            consumeEach { it.join() }
        }
    }

    fun add(block: suspend () -> Unit) {
        sendEventQueueChannel.trySend(
            sendEventScope.launch(start = CoroutineStart.LAZY) {
                block()
            }
        )
    }
}

```

Usage:

```kotlin
private val queue = CoroutineQueue(Dispatchers.IO)

queue.add { 
  // ...
}
```